nvm.sh

```diff
@@ -217,6 +217,28 @@ nvm_rc_version() {
   nvm_echo "Found '${NVMRC_PATH}' with version <${NVM_RC_VERSION}>"
 }
 
+nvm_find_package_json() {
+  local dir
+  dir="$(nvm_find_up 'package.json')"
+  if [ -e "${dir}/package.json" ]; then
+    nvm_echo "${dir}/package.json"
+  fi
+}
+
+nvm_package_json_semver_expression() {
+  export PKG_JSON_SEMVER_EXPRESSION=''
+  local PKG_JSON_PATH
+  PKG_JSON_PATH="$(nvm_find_package_json)"
+  if [ ! -e "${PKG_JSON_PATH}" ]; then
+    nvm_err "No package.json file found"
+    return 1
+  fi
+  # TODO handle error case
+  # TODO implement without node?
```

Comment:
So basically this PR is blocked because there isn't a good way to parse json in only bash. Could you explain why you don't think nvm should have node as a dependency?

I think node being a dependency would invite opportunity to nvm by letting it reach beyond the limitations of bash and into the future. Also, nvm is already built on the idea that its users will be installing node. I realize that what I'm asking for would be a pretty significant change for nvm but I think it's worth considering further.

Have you come across any other nvm enhancements that would only be possible as an extended set of features with node installed?